src/document.c

```diff
@@ -1820,6 +1820,13 @@ gboolean document_save_file_as(GeanyDocument *doc, const gchar *utf8_fname)
 	gboolean ret;
 	gboolean new_file;
 
+	// if the personne want to save as a file who dosnt exsist
+	if (doc->priv->info_bars[MSG_TYPE_RESAVE] != NULL)
```

Comment:
This should only be done when Save As is successful.

Reply:
yeah its logic i m sry so do you think i just need to move my code into the if (new_file){
} thanks you for your feedback!

Reply:
sorry for the time to response. Now i check than the file is save.